PhysicsTools/PatAlgos/plugins/PATElectronSlimmer.cc

```diff
@@ -131,7 +131,10 @@ pat::PATElectronSlimmer::produce(edm::Event & iEvent, const edm::EventSetup & iS
         if (dropRecHits_(electron)) { electron.recHits_ = EcalRecHitCollection(); electron.embeddedRecHits_ = false; }
         if (dropCorrections_(electron)) { electron.setCorrections(reco::GsfElectron::Corrections()); }
         if (dropIsolations_(electron)) { electron.setDr03Isolation(reco::GsfElectron::IsolationVariables()); electron.setDr04Isolation(reco::GsfElectron::IsolationVariables()); electron.setPfIsolationVariables(reco::GsfElectron::PflowIsolationVariables()); electron.setEcalPFClusterIso(0); electron.setHcalPFClusterIso(0); }
-        if (dropShapes_(electron)) { electron.setShowerShape(reco::GsfElectron::ShowerShape());  }
+        if (dropShapes_(electron)) { 
+          electron.setShowerShape(reco::GsfElectron::ShowerShape());  
+          electron.setExtraShowerShapes(reco::GsfElectron::ExtraShowerShapes());
```

Comment:
So, it's OK to leave the new structs as Extras? For shower shapes I am going to leave that as is in case people want to add more preshower stuff in the future.